osutil/kcmdline/kcmdline.go

```diff
@@ -441,3 +445,39 @@ func (kap *ArgumentPattern) UnmarshalYAML(unmarshal func(interface{}) error) err
 
 	return nil
 }
+
+func (kap *ArgumentPattern) marshalToString() (string, error) {
+	return fmt.Sprintf(`%s=%s`, kap.param, kap.value.String()), nil
+}
+
+func (kap *ArgumentPattern) UnmarshalBinary(data []byte) error {
+	return kap.unmarshalFromString(string(data))
+}
+
+func (kap *ArgumentPattern) MarshalBinary() ([]byte, error) {
+	str, err := kap.marshalToString()
+	if err != nil {
+		return []byte{}, err
+	}
+	return []byte(str), nil
```

Comment:
Then these can be slightly simplified
```suggestion
	return []byte(kap.marshalToString()), nil
```